panel_view.py

```diff
@@ -382,51 +379,132 @@ def update_panel_selection(active_view, current_pos, **kwargs):
 
-        draw_position_marker(panel, panel_line)
-        update_selection(panel, region)
+        nearest_error = None
+        try:
+            nearest_error, _ = min(
+                (
+                    e_d
+                    for e_d in (next_error, previous_error)
+                    if e_d and e_d[1] < SNAP
+                ),
+                key=lambda e_d: e_d[1]
+            )
+        except ValueError:
+            ...
+
+        if nearest_error:
+            nearest_errors = [
+                e
+                for e in all_errors
+                if nearest_error['region'].contains(e['region'])
+            ]
+            start = panel.text_point(nearest_errors[0]['panel_line'], 0)
+            end = panel.text_point(nearest_errors[-1]['panel_line'], 0)
+            region = panel.line(sublime.Region(start, end))
+            # regions = panel.lines(sublime.Region(start, end))
+            # regions = [sublime.Region(r.a, r.a + 1) for r in regions]
+
+            # panel.sel().clear()
+            # panel.sel().add_all(regions)
+
+            clear_position_marker(panel)
+            update_selection(panel, region)
+        else:
+            if next_error:
+                panel_line = next_error[0]['panel_line']
+            else:
+                last_error = all_errors[-1]
+                panel_line = last_error['panel_line'] + 1
+
+            start = panel.text_point(panel_line, 0)
+            region = sublime.Region(start)
+
+            draw_position_marker(panel, panel_line)
+            update_selection(panel, region)
+
+
+if False:
+    from typing import Any, Dict, List
+
+
+def mark_visible_viewport(panel, view, errors):
+    # type: (sublime.View, sublime.View, List[Dict[str, Any]]) -> None
+    KEY = 'SL.Panel.ViewportMarker'
+    KEY2 = 'SL.Panel.ViewportMarker2'
+
+    viewport = view.visible_region()
+
+    visible_errors = [e for e in errors if viewport.contains(e['region'])]
+    if visible_errors:
+        head, end = visible_errors[0], visible_errors[-1]
+        head_line = panel.text_point(head['panel_line'] - 1, 0)
+        end_line = panel.text_point(end['panel_line'], 0)
+
+        regions = [
+            sublime.Region(head_line, head_line),
+            sublime.Region(end_line, end_line)
+        ]
+        # scope = 'foo_means_forground'  # LOL
+        scope = 'region.bluish'
+        flags = (sublime.DRAW_SOLID_UNDERLINE | sublime.DRAW_NO_FILL |
+                 sublime.DRAW_NO_OUTLINE | sublime.DRAW_EMPTY_AS_OVERWRITE)
+        panel.add_regions(KEY, regions, scope=scope, flags=flags)
+
+        scope = 'comment'
+        scope = 'region.bluish'
+        flags = sublime.DRAW_NO_OUTLINE
+        head_line = panel.text_point(head['panel_line'], 0)
+        end_line = panel.text_point(end['panel_line'] + 1, 0)
+        regions = [sublime.Region(r.a, r.a + 1) for r in panel.lines(sublime.Region(head_line, end_line))]
+        panel.add_regions(KEY2, regions, scope=scope, flags=flags)
```

Comment:
Comment one of the `add_regions` lines to hide that marker. The first one adds small cursors to the left. The second one highlights the whole area with short FILL regions. Colors (scopes) can be changed as well.